# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/ref-types/externref-fastpaths.js
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/jit-test/tests/wasm/ref-types/externref-fastpaths.js
--- a/js/src/jit-test/tests/wasm/ref-types/externref-fastpaths.js	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit-test/tests/wasm/ref-types/externref-fastpaths.js	Fri Aug 04 14:06:50 2023 +0000
@@ -22,7 +22,7 @@
 // `p` will be a register argument on most platforms.
 
 function js_externref_regarg(p) {
-    if (p !== objs[index])
+    if (p !== objs[index] && !Number.isNaN(objs[index]))
         throw new Error("Bad argument at " + index);
 }
 
@@ -33,7 +33,7 @@
 // platforms as of 2019.)
 
 function js_externref_stackarg(_0, _1, _2, _3, _4, _5, _6, _7, _8, _9, p) {
-    if (p !== objs[index])
+    if (p !== objs[index] && !Number.isNaN(objs[index]))
         throw new Error("Bad argument at " + index);
 }
 
@@ -56,7 +56,7 @@
 index = 0;
 for ( let i=0; i < iter; i++ ) {
     let res = ins1.exports.run1(objs[index]);
-    if (res !== objs[index])
+    if (res !== objs[index] && !Number.isNaN(objs[index]))
         throw new Error("Bad return at " + index);
     index = (index + 1) % objs.length;
 }
@@ -64,7 +64,7 @@
 index = 0;
 for ( let i=0; i < iter; i++ ) {
     let res = ins1.exports.run2(objs[index]);
-    if (res !== objs[index])
+    if (res !== objs[index] && !Number.isNaN(objs[index]))
         throw new Error("Bad return at " + index);
     index = (index + 1) % objs.length;
 }
@@ -94,7 +94,7 @@
 index = 0;
 for ( let i=0; i < iter; i++ ) {
     let res = ins2.exports.run1();
-    if (res !== objs[index])
+    if (res !== objs[index] && !Number.isNaN(objs[index]))
         throw new Error("Bad return at " + index);
     index = (index + 1) % objs.length;
 }